### PR TITLE
issue: 1137302 Support compilation under gcc7.x

### PR DESCRIPTION
**Pull request by igor-ivanov**
These changes resolve few compilation issues observed under gcc 7.x

### CODE REVIEW

**Comment by liranoz12** [comment]
@rafi please review

File: src/stats/stats_publisher.cpp [hunk 1]
```diff
@@ -187,7 +187,11 @@ void vma_shmem_stats_open(vlog_levels_t** p_p_vma_log_level, uint8_t** p_p_vma_l
 
 	g_sh_mem_info.filename_sh_stats[0] = '\0';
 	g_sh_mem_info.p_sh_stats = MAP_FAILED;
-	sprintf(g_sh_mem_info.filename_sh_stats, "%s/vmastat.%d", safe_mce_sys().stats_shmem_dirname, getpid());
+	ret = snprintf(g_sh_mem_info.filename_sh_stats, sizeof(g_sh_mem_info.filename_sh_stats), "%s/vmastat.%d", safe_mce_sys().stats_shmem_dirname, getpid());
+	if (!((0 < ret) && (ret < (int)sizeof(g_sh_mem_info.filename_sh_stats)))) {
+                vlog_printf(VLOG_ERROR, "%s: Could not create file under %s %m\n", __func__, safe_mce_sys().stats_shmem_dirname, errno);
```
Comment (rafiw) [inline src/stats/stats_publisher.cpp]: please use tabs

File: src/vma/dev/net_device_table_mgr.cpp [hunk 2]
```diff
@@ -429,9 +429,9 @@ bool net_device_table_mgr::verify_eth_qp_creation(const char* ifname, uint8_t po
 	priv_safe_read_file(resource_path, sys_res, 1024);
 	for (int j=0; j<num_devices; j++) {
 		char ib_res[1024] = {0};
-		char ib_path[256] = {0};
```
Comment (rafiw) [inline src/vma/dev/net_device_table_mgr.cpp]: will you consider this?
```
const char tmp[] = "%s/device/resource";
char ib_path[IBV_SYSFS_PATH_MAX + sizeof(tmp)] = {0};
snprintf(ib_path, sizeof(ib_path), tmp, pp_ibv_context_list[j]->device->ibdev_path);
```

File: src/vma/proto/flow_tuple.cpp [hunk 3]
```diff
@@ -161,7 +161,7 @@ bool flow_tuple::is_3_tuple()
 void flow_tuple::set_str()
 {
 	/* cppcheck-suppress wrongPrintfScanfArgNum */
-	snprintf(m_str, STR_MAX_LENGTH, "dst:%d.%d.%d.%d:%d, src:%d.%d.%d.%d:%d, protocol:%s",
```
Comment (rafiw) [inline src/vma/proto/flow_tuple.cpp]: you can also use
"dst:%hd.%hd.%hd.%hd:%d, src:%hd.%hd.%hd.%hd:%d, protocol:%s"
and then leave the size of STR_MAX_LENGTH at 100

File: configure.ac [hunk 4]
```diff
@@ -535,20 +535,14 @@ if test "x${GCC}" = "xyes" ; then
...
-                        WNO_FLAGS=""
-                        AC_MSG_RESULT([>= 4.6])
-                else
-                        WNO_FLAGS=""
-                        AC_MSG_RESULT([< 4.6])
-                fi
-        else
-                AC_MSG_RESULT([< 4])
+        if test "${GCC_MAJOR_VER}" -ge "7" ; then
+            C_SUPPRESS_FLAGS=""
```
Comment (rafiw) [inline configure.ac]: whay is this needed?

File: configure.ac [hunk 5]
```diff
@@ -535,20 +535,14 @@ if test "x${GCC}" = "xyes" ; then
...
+            CXX_SUPPRESS_FLAGS="-Wno-deprecated"
         fi
+        AC_MSG_RESULT([${GCC_VER}])
         SHARED_FLAGS="-Wall -Wextra -Werror -Wundef -ffunction-sections -fdata-sections -Wsequence-point -pipe -Winit-self -Wmissing-include-dirs"
-        CFLAGS="$SHARED_FLAGS $CFLAGS $WNO_FLAGS"
-        CXXFLAGS="$SHARED_FLAGS -Wshadow $CXXFLAGS"
+        CFLAGS="$SHARED_FLAGS $CFLAGS $C_SUPPRESS_FLAGS"
+        CXXFLAGS="$SHARED_FLAGS -Wshadow $CXXFLAGS $CXX_SUPPRESS_FLAGS"
```
Comment (rafiw) [inline configure.ac]: would you consider this instead?
in a/src/vma/sock/sock-redirect.h
```
+#if __GNUC__ >= 7
+#define VMA_THROW(x)
+#else
+#define VMA_THROW(x) throw(x)
+#endif
int (*setsockopt) (int __fd, int __level, int __optname, __const void *__optval, socklen_t __optlen) VMA_THROW (vma_error);
```

this is more accurate.

Comment (rafiw) [reply configure.ac]: after thinking about it and reading about this how about removing the entire throw?

**Comment by igor-ivanov** [comment]
@rafiw see last update

File: src/vma/proto/flow_tuple.cpp [hunk 6]
```diff
@@ -174,7 +174,7 @@ void flow_tuple::set_str()
 void flow_tuple_with_local_if::set_str()
 {
 	/* cppcheck-suppress wrongPrintfScanfArgNum */
-	snprintf(m_str, STR_MAX_LENGTH, "dst:%d.%d.%d.%d:%d, src:%d.%d.%d.%d:%d, protocol:%s, local if:%d.%d.%d.%d",
+	snprintf(m_str, sizeof(m_str), "dst:%hd.%hd.%hd.%hd:%hd, src:%hd.%hd.%hd.%hd:%hd, proto:%s, if:%hd.%hd.%hd.%hd",
```
Comment (rafiw) [inline src/vma/proto/flow_tuple.cpp]: this should be dst:%hd.%hd.%hd.%hd**:%d**

**Comment by liranoz12** [comment]
Merged #465

**Pull request closed without merging**